### DISCUSSION TITLE
Investments and Cost Basis

### DISCUSSION THREAD

**Discussion by brooksgarrett** [apauley/hledger-flow#106] [Q&A]
I'm tracking my investments with hledger and hledger-flow. So far I have a fairly good pattern for purchases:

```
if %class SecurityTransactions &
%type (Purchase)
  description %symbol %type %longDescription
  currency
  currency1 $
  account2 equity:conversion
  amount2 -%amount1
  currency2 $
  account3 equity:conversion
  amount3 -%quantity %symbol @ $%price
  account4 assets:merrill lynch:%accountName:%symbol
  amount4 %quantity %symbol @ $%price
  comment price:%price
```

This gives me a good cost basis for `-B` reporting and also handles the conversion account so we aren't creating value from thin air.

The problem is when I have a sale I have no way of knowing at process/import time which tax lot that transaction belongs to. My first inclination is to use `post-import.journal` to simply post "correction" entries that will handle the capital gains calculations. Is this a good approach or has this been solved more elegantly by someone else?

**Comment by simonmichael** [comment]
Only part of a solution, but what about creating a [dated subaccount for the lot](https://hledger.org/investments.html#what-about-lots-) ?

**Reply by brooksgarrett** [reply]
That's a good idea to have anyway and I'll add it to my purchases but still doesn't solve matching the sale lot to the first lot due out and performing the gain/loss calculations. Maybe I could solve it in a construct script but that feels very heavy compared to just reconciling the gains in a manual post-import.

**Reply by brooksgarrett** [reply]
(Nerdy fanboi moment) And thank you for giving us hledger :)

**Reply by simonmichael** [reply]
There's no automated solution yet (with hledger) for selecting the right lot subaccount and calculating the right gain/loss at sale time - I do it semi-manually (assisted by the close command).

You are most welcome!
